sdk/clientcore/core/src/main/java/io/clientcore/core/util/ClientLogger.java

```diff
@@ -420,6 +430,27 @@ public LoggingEventBuilder addKeyValue(String key, Supplier<String> valueSupplie
             return this;
         }
 
+        /**
+         * Sets the event name for the current log event. The event name is used to query all logs
+         * that describe the same event. It must not contain any dynamic parts.
+         *
+         * @param eventName The name of the event.
+         * @return The updated {@code LoggingEventBuilder} object.
+         */
+        public LoggingEventBuilder setEventName(String eventName) {
+            this.eventName = eventName;
+            return this;
+        }
+
+        /**
+         * Logs event annotated with context.
+         */
+        public void log() {
```

Comment:
It's a bit odd to have the builder do the actual logging. Should the builder just build the logging event and then the ClientLogger does the logging?

Reply:
it's stolen from the SLF4J - https://www.slf4j.org/apidocs/org/slf4j/spi/LoggingEventBuilder.html, but I agree it's a bit weird.

If we added `LoggingEvent`, it'd result in code like

```java
LoggingEvents event = logger.atLevel().addKeyValuePair(...);
logger.log(event);
```

which is more core and extra allocation.

We can rename the `LoggingEventBuilder` to `LoggingEvent`. Let me try